File: 01_Language/01_Functions/python/hash_hmac.py
# coding: utf-8

import hmac


def hash_hmac(algo, data, key, raw_output=False):
    h = hmac.new(key, data, algo)
    if raw_output:
        return h.digest()
    else:
        return h.hexdigest()


if __name__ == '__main__':
    print(hash_hmac("ripemd160", b"The quick brown fox jumped over the lazy dog.", b"secret"))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/HashHmac.java
package com.onevgo.functions;

import cn.hutool.crypto.digest.HMac;

public class HashHmac {
    public static String hashHmac(String algo, byte[] data, byte[] key) {
        return new HMac(algo, key).digestHex(data);
    }

    public static void main(String[] args) {
        System.out.println(hashHmac("HmacRipemd160", "The quick brown fox jumped over the lazy dog.".getBytes(), "secret".getBytes()));
    }
}
